### DISCUSSION TITLE
Working with multiple projects concurrently

### DISCUSSION THREAD

**Discussion by srgsanky** [pr2502/ra-multiplex#80] [Q&A] [answered]
If I want to work with multiple projects at the same time, will ra-multiplex work?

```
                                                                                                
nvim client for project 1─────┐                           ┌──────▶ rust-analyzer for project 1  
                              │                           │                                     
                              │                           │                                     
                              │                           │                                     
                              ├─────▶ ra-multiplex  ──────┤                                     
                              │                           │                                     
                              │                           │                                     
nvim client for project 2─────┘                           └──────▶ rust-analyzer for project 2  
                                                                                                
```

In this case, will it correctly create two different rust-analyzer processes - one for each project?

**Comment by pr2502** [comment, marked as answer]
Yes, this is an explicitly supported workflow, you're meant to only run one ra-multiplex server.

The server instance is disambiguated based on the language server executable name, passed command line arguments, passed environment variables and root directory of the LSP workspace. For every new client connected ra-multiplex checks if a language server matching all these parameters is already running, if it finds one it just connects the client. If any of the parameters differ a new language server instance gets spawned and the client gets connected to it.
